src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -986,17 +986,20 @@ COptTasks::PrintMissingStatsWarning(CMemoryPool *mp, CMDAccessor *md_accessor,
 
 	if (0 < rel_stats->Size())
 	{
-		int length = NAMEDATALEN * rel_stats->Size() + 200;
-		char msgbuf[length];
-		snprintf(
-			msgbuf, sizeof(msgbuf),
-			"One or more columns in the following table(s) do not have statistics: %s",
+		CWStringDynamic *msgbuf = GPOS_NEW(mp) CWStringDynamic(mp);
```

Comment:
It is redundant to create msgbuf, because we need one string only. The wcstr string is not used without "One or more columns in the following table(s) do not have statistics: ". So it will be more optimal to begin wcstr with "One or more columns in the following table(s) do not have statistics: ". In this case the table list will not be copied from one string to another.

Reply:
I think it's a bad practice to append diagnostic messages into proper names and would leave it as is

Reply:
wcstr is used in diagnostic message only. Bad practice is writing complex code, allocating more memory and consuming more CPU time than possible. Why is my suggestion bad?

Reply:
@bandetto, Are you agree that msgbuf is redundant?

Reply:
You pose the question in a way that makes it easier for the reviewer to agree with you. The current solution is implemented in accordance with the style and technical approach of the code base. The solution you propose is based on your personal preferences, not on objective arguments. You left the question about increasing the execution time unanswered. All proposed changes should be supported by technical arguments, not personal preferences.

Reply:
> Do you agree that msgbuf is redundant?

It's redundant, but I'd choose readability here anyway, since this likely isn't a hot path that really matters for performance.